Sofa/Component/Mass/src/sofa/component/mass/MeshMatrixMass.cpp

```diff
@@ -48,13 +48,14 @@ Vector6 MeshMatrixMass<Vec3Types>::getMomentum ( const core::MechanicalParams*,
         for( int j=0 ; j<DataTypes::spatial_dimensions ; ++j ) momentum[3+j] += angularMomentum[j];
     }
 
+    const auto& edges = l_topology->getEdges();
     for(size_t i=0 ; i<l_topology->getNbEdges() ; ++i )
     {
-        unsigned v0 = l_topology->getEdge(i)[0];
-        unsigned v1 = l_topology->getEdge(i)[1];
+        const auto v0 = edges[i][0];
```

Comment:
@hugtalbot @damienmarchal What do you think about
`const Edge::value_type v0 = edges[i][0];`
instead of 
`const sofa::Index v0 = edges[i][0];`
?

Note that `Edge` is defined based on `sofa::Index`, so I prefer `sofa::Index` to `unsigned`.

Reply:
After carefull thinking I prefer sofa::Index against unsigned. 

I'm not in favor of using Edge::value_type. 
Because this is working only because Edge is inheriting value_type from fixed_array. But as Edge is specifying the value_type to be a Sofa::Index, the name value_type makes not sense as it suggest it could contains "anything". 
If encapsulating fixed_array was prefered instead of inheriting from it  would have allowed edge to have a clearly defined operator[] implemented like that: 
```cpp
        sofa::Index operator[](int){ m_data[i]; }
```
Which would have made explicit about the method behavior and what you can expect from it.

Reply:
I kind of agree with you. `Edge::value_type` does not bring much more info on the type than `auto`.